src/templates/PoolManager/FeeRewards/index.tsx

```diff
@@ -135,17 +186,189 @@ const FeeRewards = () => {
     return aggFees
   }
 
+  function handleClickToggle(event: React.ChangeEvent<HTMLInputElement>) {
+    const inputName = event.target.name
+
+    let feeData
+    if (inputName === 'depositFee' && feesData[inputName].isChecked) {
+      feeData = {
+        depositFee: {
+          isChecked: false,
+          feeRate: '0'
+        },
+        refferalFee: {
+          isChecked: false,
+          brokerCommision: 0,
+          managerShare: 0
+        }
+      }
+    } else if (inputName === 'depositFee' && !feesData[inputName].isChecked) {
+      feeData = {
+        depositFee: {
+          isChecked: true,
+          feeRate: currentDepositFee.toFixed()
+        },
+        refferalFee: {
+          isChecked: true,
+          brokerCommision: parseFloat(currentBrokerCommision.toFixed()),
+          managerShare: parseFloat(currentManagerShare.toFixed())
+        }
+      }
+    } else if (inputName === 'refferalFee' && !feesData[inputName].isChecked) {
+      feeData = {
+        ...feesData,
+        refferalFee: {
+          isChecked: true,
+          brokerCommision: parseFloat(currentBrokerCommision.toFixed()),
+          managerShare: parseFloat(currentManagerShare.toFixed())
+        }
+      }
+    } else if (inputName === 'refferalFee' && feesData[inputName].isChecked) {
+      feeData = {
+        ...feesData,
+        refferalFee: {
+          isChecked: false,
+          brokerCommision: 0,
+          managerShare: 0
+        }
+      }
+    } else {
+      feeData = {
+        ...feesData,
+        [inputName]: {
+          ...feesData[inputName],
+          isChecked: !feesData[inputName].isChecked
+        }
+      }
+    }
+
+    setFeesData(feeData)
```

Comment:
This code is too complex and hard to follow, and you are multiple times checking the same thing(`if (inputName === x)`), so I would refactor it to use a switch case and save some checks.
Also, the `isChecked` can be a variable outside the ifs, here is the refactored code:

```suggestion
const inputName = event.target.name
  let isChecked = feesData[inputName].isChecked

  let feeData
  switch (inputName) {
    case "depositFee":
      if (isChecked) {
        feeData = {
          depositFee: {
            isChecked: false,
            feeRate: '0'
          },
          refferalFee: {
            isChecked: false,
            brokerCommision: 0,
            managerShare: 0
          }
        }
      } else {
        feeData = {
          depositFee: {
            isChecked: true,
            feeRate: currentDepositFee.toFixed()
          },
          refferalFee: {
            isChecked: true,
            brokerCommision: parseFloat(currentBrokerCommision.toFixed()),
            managerShare: parseFloat(currentManagerShare.toFixed())
          }
        }
      }
      break;
    case "referralFee":
      if (isChecked) {
        feeData = {
          ...feesData,
          refferalFee: {
            isChecked: false,
            brokerCommision: 0,
            managerShare: 0
          }
        }
      } else {
        feeData = {
          ...feesData,
          refferalFee: {
            isChecked: true,
            brokerCommision: parseFloat(currentBrokerCommision.toFixed()),
            managerShare: parseFloat(currentManagerShare.toFixed())
          }
        }
      }
      break;
    default:
      feeData = {
        ...feesData,
        [inputName]: {
          ...feesData[inputName],
          isChecked: !isChecked
        }
      }
  }

  setFeesData(feeData)
```